src/Areas/Storage/Commands/DataLake/Directory/DirectoryCreateCommand.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using AzureMcp.Areas.Storage.Commands;
+using AzureMcp.Areas.Storage.Models;
+using AzureMcp.Areas.Storage.Options;
+using AzureMcp.Areas.Storage.Options.DataLake.Directory;
+using AzureMcp.Areas.Storage.Services;
+using AzureMcp.Commands.Storage;
+using AzureMcp.Services.Telemetry;
+using Microsoft.Extensions.Logging;
+
+namespace AzureMcp.Areas.Storage.Commands.DataLake.Directory;
+
+public sealed class DirectoryCreateCommand(ILogger<DirectoryCreateCommand> logger) : BaseStorageCommand<DirectoryCreateOptions>
+{
+    private const string CommandTitle = "Create Data Lake Directory";
+    private readonly ILogger<DirectoryCreateCommand> _logger = logger;
+
+    private readonly Option<string> _directoryPathOption = StorageOptionDefinitions.DirectoryPath;
+
+    public override string Name => "create";
+
+    public override string Description =>
+        """
+        Create a directory in a Data Lake file system. This command creates a new directory at the specified path
+        within the Data Lake file system. The directory path must include the file system name as the first component
+        (e.g., 'myfilesystem/data/logs' or 'myfilesystem/archives/2024'). The path supports nested structures using 
+        forward slashes (/). If the directory already exists, the operation will succeed and return the existing 
+        directory information. Returns directory metadata including name, type, and creation timestamp as JSON.
+          Required options:
+        - account-name: The storage account name
+        - directory-path: The full path including file system name and directory path
```

Comment:
Calling out the options feels redundant since it would already be included in the MCP tool input schema based on the configured options.